### DISCUSSION TITLE
Import JSON maximum 100 meetings

### DISCUSSION THREAD

**Discussion by NYSMuseum** [code4recovery/12-step-meeting-list#1088] [Q&A]
I'm syncing my meeting list with a datasource.  I know there are more than 100 meetings for my district but I'm only able to import 100 when I refresh the feed.  Is there a setting that limits the maximum number of entries in the JSON file either as what's being hosted or what I'm able to import?

**Comment by joshreisner** [comment]
huh, there's no limit, so it sounds like something is breaking during the import. does the progress bar only go a certain extent and stop? could you open the javascript console and see if there's an error there?

we had a user recently who encountered a similar limit and she was seeing a 404 error in the javascript console:
![Screen Shot 2023-03-12 at 10 21 12 AM](https://user-images.githubusercontent.com/1551689/225193107-1e7a1241-7d98-4ac2-b7db-785405d72fa2.png)

the script imports in batches of 25 meetings, apparently after 4 imports it started causing problems

**Comment by NYSMuseum** [comment]
It goes to 100%. I do not see any errors in the console.

**Reply by joshreisner** [reply]
ok that's good. what is the address you're importing from? we'll try it on our demo site to see if we have the same issue

**Comment by NYSMuseum** [comment]
http://aahmbnyo.w27.wh-2.com/district?districtNum=1

This is the link. The json only contains 100 entries so it’s not an import issue.

The only reason I’m asking about this is because meetings on our area site are not showing up in the district, even when they are listed for that district.

Go to the meetings page on the district site: https://district1.aahmbny.org/meetings/?tsml-day=0

And on the area site: https://nenyaa.org/meetings/?tsml-day=0&tsml-region=albany-county

I’ve selected Sunday which will hopefully come out through the link.  The area site has 21 meetings listed, the district has 14.


Any ideas?


On Mar 14, 2023, at 11:08 PM, Josh Reisner ***@***.***> wrote:



ok that's good. what is the address you're importing from? we'll try it on our demo site to see if we have the same issue

—
Reply to this email directly, view it on GitHub<https://github.com/code4recovery/12-step-meeting-list/discussions/1088#discussioncomment-5316642>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AECCBVX2NBQBNFK523OVV7TW4EXETANCNFSM6AAAAAAV3G3FIY>.
You are receiving this because you authored the thread.Message ID: ***@***.***>



Confidentiality Notice

This email including all attachments is confidential and intended solely for the use of the individual or entity to which it is addressed. This communication may contain information that is protected from disclosure under State and/or Federal law. Please notify the sender immediately if you have received this communication in error and delete this email from your system. If you are not the intended recipient you are notified that disclosing, copying, distributing or taking any action in reliance on the contents of this information is strictly prohibited.

**Reply by joshreisner** [reply]
i'm not sure i understand the data model here. sounds like you represent the district, and there are 100 meetings in the import and you are displaying 100 meetings, so hooray!

the area on the other hand has [105 meetings](https://nenyaa.org/wp-admin/admin-ajax.php?action=meetings&district=district-01) perhaps a few were removed from the source since they last updated their feed?

**Comment by NYSMuseum** [comment]
So maybe the area needs to update their feed? Are these meetings all populated by the area list? Sorry, I'm new to this plugin.

How are you able to see there are 105 d1 meetings on the area site? Are you just parsing out the html?

Senior Exhibitions Designer
518.474.2184
nysm.nysed.gov
________________________________

**Reply by joshreisner** [reply]
it sounds like you are both importing from a remote data source hosted at `aahmbnyo.w27.wh-2.com`. we didn't design this process with this use-case in mind. it was designed so that areas could import meetings from their districts. but i don't see why this can't work too, it just requires you both to update your feeds independently.

the link i shared in the last comment was to the area website JSON feed, which shows the number of meetings they are listing for the district. it's an easy way to count. i formed the link by using the base JSON URL `https://nenyaa.org/wp-admin/admin-ajax.php?action=meetings` and then appending `&district=district-01` which filters to just District 01

yes, best guess is they just need to refresh their import on the area website. or perhaps they added a few meetings manually, or from some other source? at any rate i would start by confirming they have clicked refresh
